src/gui/qgsmaptooledit.cpp

```diff
@@ -27,6 +27,11 @@
 QgsMapToolEdit::QgsMapToolEdit( QgsMapCanvas *canvas )
   : QgsMapTool( canvas )
 {
+  if ( mCanvas->project() )
+  {
+    connect( mCanvas->project(), &QgsProject::layersAdded, this, &QgsMapToolEdit::connectLayers );
+    connectLayers( mCanvas->project()->mapLayers().values() );  // Connect existing layers
```

Comment:
Should this also be executed when new layers get added to the project?

Reply:
Sure, in that couple of lines we cover both scenarios: first, we keep listening for new vector layers added to the project, so that as soon as they get added we connect their `editingStopped` SIGNAL; and second, if we create a new map tool edit and the project already has vector layers, we want to make sure these existing layers have their `editingStopped` SIGNALs connected.

Reply:
:see_no_evil: